src/shared/watchdog.c

```diff
@@ -117,20 +117,19 @@ int watchdog_set_timeout(usec_t timeout) {
 }
 
 usec_t watchdog_runtime_wait(void) {
-        usec_t rtwait, ntime;
 
         if (!timestamp_is_set(watchdog_timeout))
                 return USEC_INFINITY;
 
         /* Sleep half the watchdog timeout since the last successful ping at most */
         if (timestamp_is_set(watchdog_last_ping)) {
-                ntime = now(clock_boottime_or_monotonic());
+                usec_t ntime = now(clock_boottime_or_monotonic());
+
                 assert(ntime >= watchdog_last_ping);
-                rtwait = usec_sub_unsigned(watchdog_last_ping + (watchdog_timeout / 2), ntime);
-        } else
-                rtwait = watchdog_timeout / 2;
+                return usec_sub_unsigned(watchdog_last_ping + (watchdog_timeout / 2), ntime);
```

Comment:
you could even replace this line by:

```c
return usec_sub_unsigned(watchdog_last_ping + (watchdog_timeout / 2), 
                now(clock_boottime_or_monotonic());
```

Reply:
Not if we want to keep the assert right before: `assert(ntime >= watchdog_last_ping);`